iroh-net/src/dns/node_info.rs

```diff
@@ -251,17 +180,137 @@ pub(crate) fn parse_hickory_node_info_name(name: &hickory_proto::rr::Name) -> Op
     Some(node_id)
 }
 
-fn fmt_attr(label: &str, value: impl fmt::Display) -> String {
-    format!("{label}={value}")
+struct NodeAttrs {
+    node_id: NodeId,
+    attrs: BTreeMap<String, Vec<String>>,
```

Comment:
(can be a follow up) can we make the keys be an enum, for better type saftey in our code?